.github/workflows/deploy-js.yml

```diff
@@ -0,0 +1,42 @@
+# Deploy JS game to GitHub Pages
+
+on:
+  push:
+    branches: [ "main", "master" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.1.0
+        with:
+          submodules: 'true'
+      - uses: actions/setup-java@v1
+        with: {java-version: 11}
+      - name: Buid JS bundle
+        run: |
+          ./gradlew jsBrowserProductionWebpack
+      - name: Copy bundle to temp directory
+        run: |
+          mv build/distributions /tmp/docs
+          ls -la /tmp/docs
+      - run: rm -rf ./*
+      - uses: actions/checkout@v2.1.0
+        with:
+          ref: github-pages
```

Comment:
What about checking out the default branch (master/main), then executing a command to switch or create to the github-pages branch?

```
git checkout -B github-pages
```

That way this will be more resilient

Reply:
Good comment. I will try to use it.